src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -142,40 +156,40 @@
     (comidi/GET ["/resource_types/" [#".*" :rest]] request
       (master-request-handler request))
     (comidi/GET ["/status/" [#".*" :rest]] request
-      (master-request-handler request))
+      (master-request-handler request))))
 
-    ;; legacy CA routes
+(defn legacy-ca-routes
+  [ca-request-handler]
+  (comidi/routes
     (comidi/ANY ["/certificate_status/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/ANY ["/certificate_statuses/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_revocation_list/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/GET ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))
+                (ca-request-handler request))
     (comidi/PUT ["/certificate_request/" [#".*" :rest]] request
-      (ca-request-handler request))))
-
-(defn legacy-routes
-  "Creates all of the comidi routes for the master."
-  [master-request-handler ca-request-handler]
-  (comidi/routes
-    (comidi/context "/v2.0"
-      (v2_0-routes master-request-handler))
-    (comidi/context ["/" :environment]
-      (legacy-master-routes master-request-handler ca-request-handler))))
+                (ca-request-handler request))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(defn build-ring-handler
-  [master-handler master-mount master-api-version
-   ca-handler ca-mount ca-api-version]
-  (comidi/routes->handler
-    (legacy-routes
-      #(request-compatibility-wrapper
-        master-handler master-mount master-api-version %)
-      #(request-compatibility-wrapper
-        ca-handler ca-mount ca-api-version %))))
+(schema/defn ^:always-validate
+  build-ring-handler :- IFn
```

Comment:
Super minor and only as an FYI for the future, but this is one case where I think Ruth's alignment principle could be relaxed a little.  Reason being, when I fold functions with this style, I see a wall of `(schema/defn ^:always-validate ...` and no indication of the function name.  If at least the function name and optionally the return type were on the same line then this wall of folded functions would be disambiguated.